### ISSUE TITLE
Pairing the real MAC/ID address for multiple PS3 Bluetooth connected controllers

### ISSUE THREAD

**Issue by mdeguzis** [ProfessorKaos64/RetroRig#75]
@beaumanvienna , JC, so I found out tonight, after getting my second PS3 controller pairing doesn't fully work. You can check out the latest gamepads.shinc to see where I was going with this.

The problem as it stands, is the bluetooth pairing greps the MAC address from the bluetooth receiver with sixad. This is all well and good for one controller, but if you connect up two controllers (which works fine by the way), you'll notice in jstest-gtk, that each controller has its own "MAC Address" or "sub-MAC" so to speak. 

Here is the email I sent the dev of jstest-gtk:

```
How I can locate the info passed to something like jstest-gtk?
Namely I want to grab the specific device address (looks like a MAC address). each ps3 controller has it's own number in jstest-gtk.

See: http://i.imgur.com/30yjavO.png

Is there a way I can poll this information so I can efficiently set the MAC address from a set of controllers in my XBMC keymaps?

Thank you!
```

**Notice the [http://i.imgur.com/30yjavO.png](http://i.imgur.com/30yjavO.png)** link above to see what I mean.

If I can find out how to poll this information in the script, that is great, if not, we will have to work some kind of external signal out of the joystick.cpp module of the [GitHub Source tree](https://github.com/Grumbel/jstest-gtk/blob/master/src/joystick.cpp) for jstest-gtk. Some of the clues could be below that I have seen:

```
for(int i = 0; i < 32; ++i)
    {
      try
        {
          std::ostringstream str;
          str << "/dev/input/js" << i;
          Joystick joystick(str.str());

          joysticks.push_back(JoystickDescription(joystick.get_filename(),
                                                  joystick.get_name(),
                                                  joystick.get_axis_count(),
                                                  joystick.get_button_count()));
        }
      catch(std::exception& err)
        {
          // ok
        }
    }
```

I'm sure functions in that script call other pieces. 

*\* Of special note*\*  
You can find this bluetooth information by running this in a terminal window 
`cat /var/log/syslog | grep -i sixad-sixaxis` 

Maybe I can find a grep command to grap the current connected (should be the last active line) controller and save it to a variable. not sure.

So in summary, if there isn't an easy way to find these separate addresses, we'll have to address it and find some kind of solution.
